Compile error with just tzdb-concatenated

The following generates an error when trying to build jiff 0.2.13

```toml
[dependencies]
jiff = { version = "0.2.13", default-features=false, features=["tzdb-concatenated"] }
```

Results in this error:

```
error[E0599]: no method named `path` found for struct `error::Error` in the current scope
   --> /Users/eric/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/jiff-0.2.13/src/tz/db/concatenated/enabled.rs:282:62
    |
282 |         let file = File::open(path).map_err(|e| Error::io(e).path(path))?;
    |                                                              ^^^^ method not found in `Error`
    |
   ::: /Users/eric/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/jiff-0.2.13/src/error.rs:47:1
    |
47  | pub struct Error {
    | ---------------- method `path` not found for this struct

error[E0599]: no method named `path` found for struct `error::Error` in the current scope
   --> /Users/eric/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/jiff-0.2.13/src/tz/db/concatenated/enabled.rs:538:58
    |
538 |     let file = File::open(path).map_err(|e| Error::io(e).path(path))?;
    |                                                          ^^^^ method not found in `Error`
    |
   ::: /Users/eric/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/jiff-0.2.13/src/error.rs:47:1
    |
47  | pub struct Error {
    | ---------------- method `path` not found for this struct
```

I don't know if it actually makes sense to have just that one feature enabled, but if not it might be nice to `compile_error!` with an explanation.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,16 @@
 # CHANGELOG
 
+0.2.14 (TBD)
+============
+TODO
+
+Bug fixes:
+
+* [#365](https://github.com/BurntSushi/jiff/issues/365):
+Fixes a compile error in Jiff when only the `tzdb-concatenated` feature was
+enabled.
+
+
 0.2.13 (2025-05-05)
 ===================
 This release fixes a bug in a corner case where `TimeZone::following` could

--- a/scripts/test-various-feature-combos
+++ b/scripts/test-various-feature-combos
@@ -17,6 +17,7 @@ features=(
     "std tzdb-bundle-always tzdb-zoneinfo"
     "std tzdb-bundle-always logging"
     "std tzdb-bundle-always serde"
+    "std tzdb-concatenated"
 )
 for f in "${features[@]}"; do
     echo "===== FEATURES: '$f' ====="

--- a/src/error.rs
+++ b/src/error.rs
@@ -197,9 +197,7 @@ impl Error {
     ///
     /// This is a convenience routine for calling `Error::context` with a
     /// `FilePathError`.
-    ///
-    /// This is only available when the `std` feature is enabled.
-    #[cfg(feature = "tzdb-zoneinfo")]
+    #[cfg(any(feature = "tzdb-zoneinfo", feature = "tzdb-concatenated"))]
     pub(crate) fn path(self, path: impl Into<std::path::PathBuf>) -> Error {
         let err = Error::from(ErrorKind::FilePath(FilePathError {
             path: path.into(),

